packages/commonwealth/client/scripts/views/components/edit_profile.tsx

```diff
@@ -0,0 +1,414 @@
+import React, { useState, useRef } from 'react';
+import { useNavigate } from 'react-router-dom';
+import $ from 'jquery';
+import _ from 'underscore';
+
+import 'components/edit_profile.scss';
+
+import app from 'state';
+import { QuillEditorComponent } from 'views/components/quill/quill_editor_component';
+import type { QuillEditor } from 'views/components/quill/quill_editor';
+import { notifyError } from 'controllers/app/notifications';
+import {
+  NewProfile as Profile,
+  Account,
+  AddressInfo,
+  MinimumProfile,
+} from '../../models';
+import { CWButton } from '../components/component_kit/cw_button';
+import { CWTextInput } from '../components/component_kit/cw_text_input';
+import { AvatarUpload } from '../components/avatar_upload';
+import { CWSpinner } from '../components/component_kit/cw_spinner';
+import { CWText } from '../components/component_kit/cw_text';
+import { CWDivider } from '../components/component_kit/cw_divider';
+import { CWForm } from '../components/component_kit/cw_form';
+import { CWFormSection } from '../components/component_kit/cw_form_section';
+import { CWSocials } from '../components/component_kit/cw_socials';
+import type { ImageBehavior } from '../components/component_kit/cw_cover_image_uploader';
+import { CWCoverImageUploader } from '../components/component_kit/cw_cover_image_uploader';
+import { PageNotFound } from '../pages/404';
+import { LinkedAddresses } from './linked_addresses';
+
+enum EditProfileError {
+  None,
+  NoProfileFound,
+}
+
+const NoProfileFoundError = 'No profile found';
+
+type EditNewProfileProps = {
+  profileId: string;
+};
+
+export type Image = {
+  url: string;
+  imageBehavior: ImageBehavior;
+};
+
+const EditProfileComponent = (props: EditNewProfileProps) => {
+  const navigate = useNavigate();
+  const [email, setEmail] = useState('');
+  const [error, setError] = useState<EditProfileError>(EditProfileError.None);
+  const [loading, setLoading] = useState(true);
+  const [socials, setSocials] = useState<string[]>();
+  const [profile, setProfile] = useState<Profile>();
+  const [name, setName] = useState('');
+  const [avatarUrl, setAvatarUrl] = useState();
+  const [bio, setBio] = useState();
+  const [addresses, setAddresses] = useState<AddressInfo[]>();
+  const [isOwner, setIsOwner] = useState();
+  const [backgroundImage, setBackgroundImage] = useState<Image>();
+  const [displayNameValid, setDisplayNameValid] = useState(true);
+  const backgroundImageRef = useRef<Image>();
+  backgroundImageRef.current = backgroundImage;
```

Comment:
Maybe you will have a different suggestion - I am running into an issue that when the image gets uploaded/generated and returned in `CWCoverImageUploader`, the component does not get the latest set background image in state. Instead, it is referencing the previous background image. The work around here I found was to set this to the ref and use the .current value of the ref instead.

Reply:
Ok so I tried this solution - remove state and only use ref and it works great. Is this an acceptable approach?

Reply:
sorry I missed the comment. Yeah it is fine. This is the main difference between useRef and useState. They both hold the state but if you use setState, it trigers rerender. To avoid rerendering and still having the possibility to change the state, using ref and ref.current is the way to go!